libs/live-nft-react/src/hooks/useSpamTxFiltering.ts

```diff
@@ -0,0 +1,133 @@
+import { getAccountCurrency } from "@ledgerhq/coin-framework/account/helpers";
+import { BlockchainsType, BlockchainEVM } from "@ledgerhq/live-nft/supported";
+import { NFTResource } from "@ledgerhq/live-nft/types";
+import {
+  Operation,
+  AccountLike,
+  DailyOperations,
+  NFTCollectionMetadataResponse,
+} from "@ledgerhq/types-live";
+import { useMemo } from "react";
+import { useNftCollectionMetadataBatch } from "../NftMetadataProvider";
+
+type AccountMap = Record<string, AccountLike>;
+
+type NftOp = {
+  contract: string;
+  currencyId: string;
+};
+
+type NftOperations = Record<string, NftOp>;
+
+export function useNftOperations({
+  accountsMap,
+  groupedOperations,
+  spamFilteringTxEnabled,
+}: {
+  accountsMap: AccountMap;
+  groupedOperations?: DailyOperations;
+  spamFilteringTxEnabled: boolean;
+}) {
+  return useMemo(
+    () =>
+      spamFilteringTxEnabled
+        ? groupedOperations?.sections
+            .flatMap(section => section.data)
+            .reduce((acc, operation) => {
+              if (operation.type === "NFT_IN") {
+                const account = accountsMap[operation.accountId];
+                const contract = operation.contract || "";
+                acc[contract] = {
+                  contract,
+                  currencyId: getAccountCurrency(account).id,
+                };
+              }
+              return acc;
+            }, {} as NftOperations) || ({} as NftOperations)
+        : ({} as NftOperations),
```

Comment:
Can't we get rid of all the `as` by typing the functions passed to the useMemo and the reducer:
```ts
return useMemo(
  (): NftOperations =>
```
```ts
.reduce((acc: NftOperations, operation): NftOperations => {
```
It should also make the ` || {}` part not needed:
```suggestion
            }, {})
        : {},
```